include/vptr/virtual_ptr.hpp

```diff
@@ -35,13 +35,21 @@
 
 #include <CL/sycl.hpp>
 
+#ifdef SYCL_IMPLEMENTATION_ONEAPI
+#if __SYCL_COMPILER_VERSION >= 20220812
+#define NO_CL_NAMESPACE
+#endif
+#endif
+
 #include <cstddef>
 #include <queue>
 #include <set>
 #include <stdexcept>
 #include <map>
 
+#ifndef NO_CL_NAMESPACE
 namespace cl {
+#endif
 namespace sycl {
 namespace codeplay {
```

Comment:
As we'd previously discussed, would you consider changing this to some other namespace, like `vptr` or something? It doesn't really need and likely shouldn't be in the `sycl` namespace at all. Users can remain on the old version if they don't want to change their code, meanwhile we can update SYCL-BLAS when this is merged.

Reply:
The namespace `vptr` is now used instead of `cl::sycl::computecpp`. The tests and samples have been updated to reflect this, and the change still achieves the desired compatability with SYCL-BLAS and DPC++.